pallets/runtime/tests/src/utility_test.rs

```diff
@@ -228,3 +231,92 @@ fn _relay_unhappy_cases() {
         Error::InvalidNonce
     );
 }
+
+#[test]
+fn batch_secondary_with_permissions_works() {
+    ExtBuilder::default()
+        .build()
+        .execute_with(batch_secondary_with_permissions);
+}
+
+fn batch_secondary_with_permissions() {
+    System::set_block_number(1);
+    let alice = User::new(AccountKeyring::Alice).balance(1_000);
+    let alice_key = AccountKeyring::Alice.public();
+    let alice_origin = Origin::signed(alice_key);
+    let alice_did = alice.did;
+    let bob_key = AccountKeyring::Bob.public();
+    let bob_origin = Origin::signed(bob_key);
+    let bob_signer = Signatory::Account(bob_key);
```

Comment:
Prefer `alice.signer()` and `alice.did` etc. at use sites instead rather than binding here.

Reply:
"Signer" is a not the right choice of name for an origin. A "signer" should be of type `Signatory`. I'll correct the method name.